src/test/java/htsjdk/samtools/util/IOUtilUnitTest.java

```diff
@@ -116,12 +108,12 @@ public void testGetCanonicalPath() throws IOException {
             Assert.assertEquals(IOUtil.getFullCanonicalPath(f), actual.getCanonicalPath());
         }
 
-        actual.delete();
-        symlink.delete();
-        lnToActual.delete();
-        lnToSymlink.delete();
-        lnDir.delete();
-        tmpDir.delete();
+        actual.deleteOnExit();

```

Comment:
Are you trying to silence the "Not check of returned boolean value" Intellij warning by chaning from delete to deleteOnExit? I would say that if it is not a warning as compiled by gradle one should not care about that and other warnings... it shouldn't influence the code (you should silence it in Intellij if it bother you). 

In any case it seem more logic to call deleteOnExit right after you create the File as a failure in the middle would prevent the .delete() to be reached... I think you can call deleteOnExit on a not yet existent file.

Reply:
If you want to silence those delete() warning you could make them part of the test by calling ```Assert.assertTrue(x.delete())``` if they are supposed exist after running the test code. If the contrary you could call ```Assert.assertFalse(x.delete())``` just the same.

Reply:
Yeah, those yellows in intellij bother me.   That's a good point about moving them to be next to creation.  I think I'm just going to revert the change and leave them with delete though.